src/libusb/device-libusb.h

```diff
@@ -0,0 +1,44 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "usb/usb-types.h"
+#include "messenger-libusb.h"
+#include "usb/usb-device.h"
+
+#include <string>
+#include <regex>
+#include <sstream>
+#include <mutex>
+
+#include <libusb.h>
+
+namespace librealsense
+{
+    namespace platform
+    {
+        class usb_device_libusb : public usb_device, public std::enable_shared_from_this<usb_device_libusb>
+        {
+        public:
+            usb_device_libusb(libusb_device* device, libusb_device_descriptor desc);
+            virtual ~usb_device_libusb() {}
+
+            virtual const usb_device_info get_info() const override { return _infos[0]; }
+            virtual const std::vector<usb_device_info> get_subdevices_infos() const override { return _infos; }
+            virtual const rs_usb_interface get_interface(uint8_t interface_number) const override { return _interfaces.at(interface_number); }
```

Comment:
This will throw if the index is not in the map

Reply:
get_interface(uint8_t interface_number) will be removed, leaving only get_interfaces()